specs/rollup-node.md

```diff
@@ -347,6 +351,25 @@ block forward each step, until there is an insufficient number of L1 blocks left
 
 > **TODO** specify genesis block
 
+## L2 Output RPC method
+
+The Rollup node has its own RPC method, `optimism_outputAtBlock` which returns the
+a 32 byte hash corresponding to the [SSZ] encoded [L2Output](./proposals.md#l2-output-commitment-construction).
+
+[SSZ]: https://github.com/ethereum/consensus-specs/blob/dev/ssz/simple-serialize.md
+
+### Output Method API
+
+The input and return types here are as defined by the [engine API specs][engine-structures]).
+
+[engine-structures]: https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#structures
+
+- method: `optimism_outputAtBlock`
+- params:
+  1. `QUANTITY` - L2 integer block number, or the strings `"safe"`, `"latest"`, or `"pending"`
+- returns:
+  1. `DATA` - The 32 byte output root
+
```

Comment:
I think that this should either accept a "type" param or return JSON that includes data and type so that we can upgrade the commitments more easily, cc @protolambda

Reply:
We could embed the type enum in the first byte and just use typed hashes

Reply:
How is this?

```
- method: `optimism_outputAtBlock`
- params:
  1. `TYPE` - A versioning enum which describes the data structure used to generate the output root. Defaults to the latest version, which is currently '0'
  1. `QUANTITY` - L2 integer block number, or the strings `"safe"`, `"latest"`, or `"pending"`
- returns:
  1. `DATA` - The 32 byte output root
```

Also, should it return just the output root, or all the leaves? 

Should you also be able to get a proof from an `optimism_` method on the rollup node?

Reply:
Shouldn't the type be a return value? The sequencer choses the type of the commitment he submits after all, the user doesn't get a say.

The submitter should also specify this output type. We may want to validate the output type in the contract (i.e. probably just check that it is the latest known type) — although that will also naturally be done in the rollup-node / minigeth (i.e. minigeth will always derive the correct ouptut root type, so submitting an old output root type will be contestable via a fault proof). Still a bit of redundant security doesn't hurt.

Reply:
Yes, you're totally right.

Reply:
A few thoughts with this design:
- The rollup node will be responsible for knowing what version to return. Since it can speak to the execution client, it has the context of the chain and can alter the version at a specific height
- The output submitter can be dumb and not need to understand what the types mean
- The contract needing to know the type means that it'll consume extra gas when submitting L2 outputs, do we need logic that says: `if blocknumber == x; then update version` - how exactly do we govern that? The contracts aren't specified to be behind delegatecall proxies right now